src/docfx/build/toc/parser/MarkdownTocMarkup.cs

```diff
@@ -124,9 +124,7 @@ TableOfContentsItem GetItem(HeadingBlock block)
                 {
                     if (!string.IsNullOrEmpty(linkInline.Url))
                     {
-                        // TODO: check why line starts from 0
-                        // TODO: check why column info is missing
-                        currentItem.Href = new SourceInfo<string>(linkInline.Url, new Range(block.Line, block.Column));
+                        currentItem.Href = new SourceInfo<string>(linkInline.Url, linkInline.ToRange());
```

Comment:
Add `UsePreciseSourceLocation` to TOC markdown pipeline setup for this line info to work